# HG changeset patch
# User Pascal Dupuis <Pascal.Dupuis@uclouvain.be>
# Date 1296380290 18000
#      Sun Jan 30 04:38:10 2011 -0500
# Branch release-3-4-x
# Node ID 1b71befc2f409ed67697aa8754f4e466636e34d0
# Parent  3eb4049b5393e043cb3602b941b89521f7944b6a
oct-parse.yy (class stdio_stream_reader): disallow copying

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2011-01-30  Pascal Dupuis  <Pascal.Dupuis@worldonline.be>
+
+	* oct-parse.yy (class stdio_stream_reader): Disallow copying.
+
 2011-01-30  John W. Eaton  <jwe@octave.org>
 
 	* graphics.cc (gnuplot_toolkit::redraw): Call __gnuplot_drawnow__,
 	not gnuplot_drawnow.
 	(gnuplot_toolkit::print_figure): Likewise.
 
 2011-01-30  John W. Eaton  <jwe@octave.org>
 
diff --git a/src/oct-parse.yy b/src/oct-parse.yy
--- a/src/oct-parse.yy
+++ b/src/oct-parse.yy
@@ -3320,16 +3320,22 @@ public:
     if (c == '\n')
       input_line_number--;
 
     return ::ungetc (c, f);
   }
 
 private:
   FILE *f;
+
+  // No copying!
+
+  stdio_stream_reader (const  stdio_stream_reader&);
+
+  stdio_stream_reader & operator = (const  stdio_stream_reader&);
 };
 
 static bool
 skip_white_space (stream_reader& reader)
 {
   int c = 0;
 
   while ((c = reader.getc ()) != EOF)
